src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -768,18 +768,136 @@ protected bool IsAlive
+        /// <returns>The input elements with their geometry unjoined.</returns>
+        public IEnumerable<Element> UnjoinGeometry(Element otherElement)
+        {
+            if (!JoinGeometryUtils.AreElementsJoined(Document, this.InternalElement, otherElement.InternalElement))
+                throw new InvalidOperationException(Properties.Resources.NotJoinedElements);
+
+            TransactionManager.Instance.EnsureInTransaction(Document);
+            JoinGeometryUtils.UnjoinGeometry(
+                        Document,
+                        this.InternalElement,
+                        otherElement.InternalElement);
+            TransactionManager.Instance.TransactionTaskDone();
+            return new List<Element>() { this, otherElement };
+        }
+
+        /// <summary>
+        /// Unjoins the geometry of all elements from each other if they are joined.
+        /// This performs only one transaction in Revit.
+        /// </summary>
+        /// <param name="elements">List of elements to unjoin from each other</param>
+        /// <returns>All input Elements, with their geometry now unjoined from each other.</returns>
+        public static IEnumerable<Element> UnjoinAllGeometry(List<Element> elements)
+        {
+            TransactionManager.Instance.EnsureInTransaction(Document);
+            for (int i = 0; i < elements.Count; i++)
+            {
+                List<Element> joinedElements = JoinGeometryUtils.GetJoinedElements(Document, elements[i].InternalElement)
+                                                                .Select(id => Document.GetElement(id).ToDSType(true))
+                                                                .ToList();
+                if (joinedElements.Count <= 0)
+                    continue;
+
+                for (int j = 0; j < joinedElements.Count; j++)
+                {
+                    JoinGeometryUtils.UnjoinGeometry(
+                        Document,
+                        elements[i].InternalElement,
+                        joinedElements[j].InternalElement);
+                }
+            }
+            TransactionManager.Instance.TransactionTaskDone();
+            return elements;
+        }
+        
+        /// <summary>
+        /// Switch the order in which the geometry of two elements is joined. If the order is already as desired, elements are not affected. 
+        /// Note that changing this will affect both 3D and 2D views and that it is not the recommended way of managing wall joins.
+        /// </summary>
+        /// <param name="cuttingElement">The element that should be cutting the other element</param>
+        /// <param name="otherElement">The other element that is being cut by the cuttingElement</param>
+        /// <returns>Joined elements</returns>
+        public static IEnumerable<Element> SwitchGeometryJoinOrder(Element cuttingElement, Element otherElement)
```

Comment:
Is this right?
Before this node is executed, `cuttingElement` cut `otherElement`, and after executed, `otherElement` will cut `cuttingElement`.
Run the script again, the order will be changed again?

Reply:
Line 867 checks if `cuttingElement` is already the cutting Element, if it is it returns the original order if not it will switch the order.

Reply:
We also have tests for this here : https://github.com/SHKnudsen/DynamoRevit/blob/8b2376672c96d3b904b16d057cba2345f82b3d9f/test/Libraries/RevitNodesTests/Elements/ElementTests.cs#L529-L532

```csharp
            // Elements already in the wanted join order
            IEnumerable<Element> orderedElements = Element.SwitchGeometryJoinOrder(cuttingFraming, cutFraming);
            List<int> orderedElementIds = orderedElements.Select(elem => elem.Id).ToList();
            CollectionAssert.AreEqual(unchangedOrder, orderedElementIds);
```

Reply:
OK. Got it.

Reply:
So it is set join order, not switch join order?

Reply:
i guess technically it is - would you like us to rename it to `SetGeometryJoinOrder` ?

Reply:
Agree.

Reply:
node is now renamed